### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by girma336**
### Limit access to web app resources based on authorization rules

- [ ]  Added authorization for blog app

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    # can :manage, :all
+
+    user ||= User.new # guest user (not logged in)
+    if user.is? :admin
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, Post, author_id: user.id
+      can :create, Comment, :all
+    end
```
Comment (MeqdamAlqudah) [inline app/models/ability.rb]: - [ ] Please good Job by mentioning the post and comment in the ability but as it was requested by the project requirements, please the user  can manage all the posts and comments if they are his own comments and posts so consider changing create to manage currently the user can just create them

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,11 +1,13 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!
   def index
```
Comment (MeqdamAlqudah) [inline app/controllers/posts_controller.rb]: - [ ] Please use CanCanCan for Post authorization in Post Controller, you can do this by adding this line to postController `load_and_authorize_resource`

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @girma336 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image
## Highlights
:+1:  admin authorization 
:+1: controllers structure

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- changing the :create to :manage in the ability file , and adding author: user at the same time

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @girma336, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 

- cancancan role-based authorization has effectively been configured :+1: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**